test-cypress/integration/explore/similarity.spec.js

```diff
@@ -0,0 +1,299 @@
+                          ":score ?score. \n" +
+                          "OPTIONAL { ?result <http://dbpedia.org/ontology/birthPlace> ?birthDate . }}";
+
+	before(function () {
+	//Sets the default timeout to 15s. so there's plenty of time for index building.
+	//defaultCommandTimeout is set per spec so it won't affect other tests.
+	    Cypress.config('defaultCommandTimeout', 15000);
+		cy.visit('/repository');
+		cy.wait(1000);
+		cy.createNewRepo(repoId).wait(2000);
+		cy.setRepoDefault(repoId);
+		cy.selectRepo(repoId);
+		cy.navigateToPage('Import', 'RDF');
+		cy.chooseServerFilesTab();
+    	cy.selectDistinctFileToImport(fileToImport);
+    	cy.importChangingSettings().wait(2000);
+    	cy.verifyMessageOccurs(messageSuccess);
+		cy.visit('/similarity').wait(1000);
+	});
+
+
+//	beforeEach(function () {
+//		// Go to Similarity page
+//		cy.visit('/similarity').wait(1000);
+//	});
+
+	after(function () {
+	    //Delete repository
+		cy.navigateToPage('Setup', 'Repositories');
+		cy.deleteRepo(repoId);
+	});
+
+	let checkSimilarityPageDefaultState = function () {
+	    //Verify there are no indexes
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+        cy.get('div[class="card mb-2"]').should('be.visible').should('contain', 'No Indexes');
+	};
+
+	let pressCreateNewIndexButton = function () {
+        cy.get('div[class="clearfix"] span[class="icon-plus"]').click({force:true});
+        cy.wait(500);
+    };
+
+	let setIndexName = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        cy.get('div [class="input-group input-group-lg"] input').type(indexName);
+        cy.wait(500);
+        //cy.get('div [class="input-group input-group-lg"] input').should('contain', indexName);
+    };
+
+	let expandSimilarityIndexOptionsMenu = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        cy.get('div button[class="btn btn-link btn-sm mt-1 px-0 collapsed"]').should("be.visible");
+        cy.get('div button[class="btn btn-link btn-sm mt-1 px-0 collapsed"]').click();
+	};
+
+	let addBuildParam = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        expandSimilarityIndexOptionsMenu();
+        cy.get('#indexParameters').type(buildParam);
+        //Should add functionality to check whether the parameter is indeed passed to the field
+        cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click().wait(500);
+
+    };
+
+	let addStopWord = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        expandSimilarityIndexOptionsMenu();
+        cy.get('div[class="col-md-8"] input').eq(0).type(stopWord);
+        cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click().wait(500);
```

Comment:
`cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click()` this is used 4 times in different functions. We can have it in separate function and reuse it. Also why here we `wait(500)` but in the other places we don't?!

Reply:
The wait is not needed in this operation so I removed it. Added function collapseSimilarityIndexOptionsMenu() as well.